header_test.go

```diff
@@ -50,3 +50,38 @@ func TestEmptyContentType(t *testing.T) {
 		t.Errorf("Expected media type %q but got %q", mediaType, gotMediaType)
 	}
 }
+
+func TestKnownCharset(t *testing.T) {
+	var h Header
+
+	h.Set("Subject", "=?ISO-8859-1?B?SWYgeW91IGNhbiByZWFkIHRoaXMgeW8=?=")
+
+	fields := h.Fields()
+	if !fields.Next() {
+		t.Error("Expected to be able to advance to first header item")
+	}
+
+	_, err := fields.Text()
+	if err != nil {
+		t.Error("Expected no error when decoding header key of known charset, but got: ", err)
+	}
+}
+
+func TestUnknownCharset(t *testing.T) {
+	var h Header
+
+	h.Set("Subject", "=?ISO-8859-2?B?dSB1bmRlcnN0YW5kIHRoZSBleGFtcGxlLg==?=")
```

Comment:
If we add support for this charset in the future, this will break the test. Maybe we can use an obviously-wrong charset instead, like `INVALIDCHARSET`?

Reply:
agreed, fixed